include/pros/adi.hpp

```diff
@@ -94,15 +118,29 @@ class ADIAnalogIn : private ADIPort {
 	 * reached:
 	 * ENXIO - The given value is not within the range of ADI Ports
 	 *
-	 * \param port
+	 * \param adi_port
 	 *        The ADI port number (from 1-8, 'a'-'h', 'A'-'H') to configure
-	 * \param type
-	 *        The configuration type for the port
 	 *
 	 * \return 1 if the operation was successful or PROS_ERR if the operation
 	 * failed, setting errno.
 	 */
-	ADIAnalogIn(std::uint8_t port);
+	explicit ADIAnalogIn(std::uint8_t adi_port);
+
+	/**
+	 * Configures an ADI port on an adi expander to act as an Analog Input.
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of ADI Ports
+	 *
+	 * \param port_pair
+	 *        The pair of the smart port number (from 1-22) and the
+	 * 		  ADI port number (from 1-8, 'a'-'h', 'A'-'H') to configure
+	 *
+	 * \return 1 if the operation was successful or PROS_ERR if the operation
+	 * failed, setting errno.
+	 */
+	ADIAnalogIn(ext_adi_port_pair_t port_pair);
```

Comment:
What happened to `explicit` here? I noticed you added them in but then took them out in a later commit. Same with others below.

Reply:
if you put explicit on the constructors with port_pair then when you initialize the objects, you have to cast them or create a variable with the correct type first.  When you just do object({{smart, adi}}) it doesn’t automatically convert to ext_adi_port_pair_t, which causes the program to not work.

Reply:
Humm, that's strange. That's not how I remember explicit constructors working, I thought explicit would prevent something from being converted to `ADIAnalogIn` without explicit type. For example `ADIAnalogIn i = 1` does not automatically convert `1` to `ADIAnalogIn` for copy construction. However, I don't think that's the case here, the problem you described is it not being able to create a `ext_adi_port_pair_t` from `{smart, adi}`, and I would think that's the fault of explicit pair constructors. Maybe I'm wrong, but try removing a pair of curly brackets, like `object({smart, adi})` instead of `object({{smart, adi}})`.

Reply:
You were right, changing from {{smart, adi}} to {smart, adi} worked.  I have re-added the explicit

Reply:
Huh, cool. I think what was maybe happening was that it was implicitly converting the inner level of braces into an `ADIAnalogIn` for the copy constructor. It was basically implicitly doing this: `ADIAnalogIn(ADIAnalogIn({smart, adi}))`, so when you made the constructor explicit it broke that.